app/services/recovery_code_generator.rb

```diff
@@ -1,7 +1,7 @@
 class RecoveryCodeGenerator
   attr_reader :user_access_key
 
-  def initialize(user, length: 16)
+  def initialize(user, length: 12)
```

Comment:
make `12` a constant or reference Figaro env val so it is a less magical number?